pgmpy/models/DoubleMLRegressor.py

```
@@ -0,0 +1,188 @@
+"""flake8: noqa"""
+
+from typing import Any, List, Optional, Sequence, Tuple
+
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator, RegressorMixin, clone
+from sklearn.utils.validation import check_is_fitted, check_X_y
+
+
+class DoubleMLRegressor(BaseEstimator, RegressorMixin):
+    def __init__(
+        self,
+        dag: Any,
+        estimator_g: Any,
+        estimator_m: Optional[Any] = None,
+        adjustment_set: Optional[Sequence[str]] = None,
+        treatment_col: Optional[str] = None,
+        n_folds: int = 5,
+        random_state: Optional[int] = None,
+    ):
+        self.dag = dag
+        self.estimator_g = estimator_g
+        self.estimator_m = estimator_m
+        self.adjustment_set = adjustment_set
+        self.treatment_col = treatment_col
+        self.n_folds = n_folds
+        self.random_state = random_state
+
+    def _more_tags(self):
+        return {"requires_y": True, "no_sparse_input": True}
+
+    # ---------- small helpers ----------
+    def _as_dataframe(self, X) -> pd.DataFrame:
+        if isinstance(X, pd.DataFrame):
+            return X.copy()
+        arr = np.asarray(X)
+        if arr.ndim == 1:
+            arr = arr.reshape(-1, 1)
+        cols = [f"x{i}" for i in range(arr.shape[1])]
+        return pd.DataFrame(arr, columns=cols)
+
+    def _resolve_treatment_and_adjustments(
+        self, df: pd.DataFrame
+    ) -> Tuple[str, List[str]]:
+        # treatment priority: explicit -> dag.exposure -> first column
```

Not sure I agree with this treatment of roles - I think we should only support DAG roles. Passing it explicitly overrides the DAG, so they could be mismatched and the first col approach if there is no DAG roles is a potential source of a silent failure. Current fallback `treat = df.columns[0]` is dangerous. IMO, first validate that X is a pandas DataFrame and that all role-referenced nodes exist in X.columns, and throw an error otherwise. This will avoid silent misalignment.